src/DssSpell.t.sol

```diff
@@ -2211,66 +2287,90 @@ contract DssSpellTest is DSTest, DSMath {
+        assertEq(dai.balanceOf(MKT_MULTISIG)    - prevDaiMkt, amountMkt * WAD);
+        assertEq(dai.balanceOf(GOV_MULTISIG)    - prevDaiGov, amountGov * WAD);
+        assertEq(dai.balanceOf(RWF_MULTISIG)    - prevDaiRwf, amountRwf * WAD);
+        assertEq(dai.balanceOf(RISK_CU_EOA)     - prevDaiRisk, amountRisk * WAD);
+        assertEq(dai.balanceOf(PE_MULTISIG)     - prevDaiPe, amountPe * WAD);
+        assertEq(dai.balanceOf(ORA_MULTISIG)    - prevDaiOra, amountOra * WAD);
+        assertEq(dai.balanceOf(ORA_ER_MULTISIG) - prevDaiOraEr, amountOraEr * WAD);
 
-    function onFlashLoan(
-        address initiator,
-        address token,
-        uint256 amount,
-        uint256 fee,
-        bytes calldata
-    ) external returns (bytes32) {
-        assertEq(initiator, address(this));   
-        assertEq(token, address(dai));
-        assertEq(amount, 1 * MILLION * WAD);
-        assertEq(fee, 500 * WAD);
-
-        dai.approve(msg.sender, 1_000_500 * WAD);
-
-        return keccak256("ERC3156FlashBorrower.onFlashLoan");
     }
 
-    function onVatDaiFlashLoan(
-        address initiator,
-        uint256 amount,
-        uint256 fee,
-        bytes calldata
-    ) external returns (bytes32) {
-        assertEq(initiator, address(this));   
-        assertEq(amount, 1 * MILLION * RAD);
-        assertEq(fee, 500 * RAD);
-
-        dai.approve(address(daiJoin), 500 * WAD);
-        daiJoin.join(address(this), 500 * WAD);
-        vat.move(address(this), msg.sender, 1_000_500 * RAD);
-
-        return keccak256("VatDaiFlashBorrower.onVatDaiFlashLoan");
-    }
+    function testClawBackMultisigFunds() public {
+        vote(address(spell));
+        spell.schedule();
+        castPreviousSpell();
+        hevm.warp(spell.nextCastTime());
+        spell.cast();
+        assertTrue(spell.done());
 
+        uint256 daiBal = dai.balanceOf(pauseProxy);
+
+        uint256[5] memory allowancePre = GnosisAllowanceModule(ALLOWANCE_MODULE).getTokenAllowance(
+            ORA_MULTISIG, pauseProxy, address(dai)
+        );
+
+        assertEq(allowancePre[0], 10_000_000_000 * WAD); // amount
+        assertEq(allowancePre[1], 0);                    // spent
+        assertEq(allowancePre[2], 0);                    // resetTimeMin
+        assertEq(allowancePre[3], 27085100);             // lastResetMin
+        assertEq(allowancePre[4], 1);                    // nonce
+
+        uint256 castTime = block.timestamp + 30 days;
+        ClawSpell clawSpell = new ClawSpell();
+        vote(address(clawSpell));
+        clawSpell.schedule();
+        hevm.warp(castTime);
+        clawSpell.cast();
+
+        assertEq(dai.balanceOf(pauseProxy), daiBal + 10**18);
+
+        uint256[5] memory allowancePost = GnosisAllowanceModule(ALLOWANCE_MODULE).getTokenAllowance(
+            ORA_MULTISIG, pauseProxy, address(dai)
+        );
+
+        assertEq(allowancePost[0], allowancePre[0]);            // amount
```

Comment:
We should have an assert checking the actual value of the allowance is what I meant--so that we know it's enough for e.g. a year of payments.

Reply:
I added that in suggestion - see above

Reply:
I think the test you're looking for is on ln 2350

Reply:
Modified this a little to just assert that it's GT 12 months worth of payments.